src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -56,12 +56,18 @@ private Security(InstrumentationGateway instrumentationGateway = null, IPowerWaf
 
                 _instrumentationGateway = instrumentationGateway ?? new InstrumentationGateway();
 
-                _powerWaf = powerWaf ?? (Enabled ? new PowerWaf() : new NullPowerWaf());
-                _agentWriter = agentWriter ?? new AgentWriter();
-
                 if (Enabled)
                 {
-                    _instrumentationGateway.InstrumentationGetwayEvent += InstrumentationGateway_InstrumentationGetwayEvent;
+                    _powerWaf = powerWaf ?? PowerWaf.Initialize();
+                    if (_powerWaf != null)
+                    {
+                        _agentWriter = agentWriter ?? new AgentWriter();
+                        _instrumentationGateway.InstrumentationGatewayEvent += InstrumentationGatewayInstrumentationGatewayEvent;
+                    }
+                    else
+                    {
+                        Enabled = false;
+                    }
                 }
             }
             catch (Exception ex)
```

Comment:
Should probably set `Enabled = false` in this catch block too.

Reply:
good point!